keel-deliveryconfig-plugin/src/main/kotlin/com/netflix/spinnaker/keel/deliveryconfig/resource/DeliveryConfigHandler.kt

```diff
@@ -35,39 +42,98 @@ class DeliveryConfigHandler(
     "deliveryConfig:${spec.name}"
   )
 
+  private val resourcePersister: ResourcePersister by lazy { resourcePersisterProvider.get() }
+  private val resourceRepository: ResourceRepository by lazy { resourceRepositoryProvider.get() }
+
+  private val DeliveryConfig.isResolved : Boolean
+    get() = this.deliveryEnvironments.all { env ->
+      env.packageRef.resourceMetadata != null
+        && env.targets.all { it.resourceMetadata != null}
+    }
+
   override fun current(resource: Resource<DeliveryConfig>): DeliveryConfig? =
-    runBlocking {
-      log.debug("Fetching delivery ${resource.spec.name} from front50")
-      try {
-        val delivery = front50Service.deliveryById(resource.spec.name).await()
-        log.debug("Fetched $delivery")
-        DeliveryConfig(delivery.id, delivery.application)
-      } catch (e: HttpException) {
-        if (!e.isNotFound) {
-          log.error("Fetching delivery from front50 failed", e)
-          throw e
-        }
-        null
+    if (resource.spec.isResolved) {
```

Comment:
ensure that if any of the sub resources components in the spec are not yet being managed (don't have populated metadata) that we resubmit them. currently does this dumbly by just bombing out with 'current state is null' and could potentially be smarter, but the upsert doesn't look at the diff at all anyway so I think this is ok